src/main/scala/beam/replanning/ModeIterationPlanCleaner.scala

```diff
@@ -0,0 +1,36 @@
+package beam.replanning
+import beam.sim.config.BeamConfig
+import com.google.inject.Inject
+import com.typesafe.scalalogging.LazyLogging
+import org.matsim.api.core.v01.Scenario
+import org.matsim.api.core.v01.population._
+import org.matsim.core.controler.events.IterationStartsEvent
+import org.matsim.core.controler.listener.IterationStartsListener
+
+import scala.collection.JavaConverters._
+
+class ModeIterationPlanCleaner @Inject()(config: BeamConfig, scenario: Scenario)
+    extends IterationStartsListener
+    with LazyLogging {
+
+  private val clearModeIteration: Int = config.beam.replanning.clearModesAtStartOfIteration.atIteration
+  private val clearModes: Set[String] =
```

Comment:
Shall it be lowercased too?

Reply:
I don't think so - the configuration should use the same modes as we have at population. Though, we can revalidate them if it needs.

Reply:
But I see `toLowerCase` here: https://github.com/LBNL-UCB-STI/beam/pull/2664/files#diff-687b85be0b3336fce27542d298c6b368R31
I'm worried if someone copy the mode as CAR, it won't work.

Reply:
Oh I see. Ok, I will add this in a moment